extensions/blocks/story/player/utils.js

```diff
@@ -0,0 +1,54 @@
+export async function waitMediaReady( mediaElement ) {
+	const elementTag = mediaElement.tagName.toLowerCase();
+	if ( 'img' === elementTag ) {
+		if ( mediaElement.complete ) {
+			return;
+		}
+		await new Promise( resolve => {
+			mediaElement.addEventListener( 'load', resolve, { once: true } );
+		} );
+	} else if ( 'video' === elementTag || 'audio' === elementTag ) {
+		if ( mediaElement.HAVE_ENOUGH_DATA === mediaElement.readyState ) {
+			return;
+		}
+		await new Promise( resolve => {
+			mediaElement.addEventListener( 'canplaythrough', resolve, { once: true } );
+			mediaElement.load();
+		} );
+	}
+}
+
+export const fullscreen = {
+	enabled() {
+		return (
+			document.fullscreenEnabled ||
+			document.webkitFullscreenEnabled ||
+			document.mozFullScreenEnabled ||
+			document.msFullscreenEnabled
+		);
+	},
+	element() {
```

Comment:
`enabled` and `element` seem to do the same thing?

Reply:
Oh wait no they don't! I wonder if there's a way to refactor this file to be more DRY...